### ISSUE TITLE
Client connection can hang after Mirror request has been aborted

### ISSUE THREAD

**Issue by gavin-norman-sociomantic** [sociomantic-tsunami/dhtnode#13]
Reproduction steps (dhtnode and dhtproto neo-alpha-3):
1. Start a dhtnode with some data in a channel.
2. Start a client which does a Mirror request.
3. During the initial refresh cycle, SIGINT the client.
4. Upon restarting the client, the client hangs after the `this.dht.blocking.waitAllHashRangesKnown();` call exits.

**Comment by gavin-norman-sociomantic** [comment]
On the second run, it gets to https://github.com/sociomantic-tsunami/dhtproto/blob/neo/src/dhtproto/node/neo/request/Mirror.d#L989, suspends, but is not resumed. I guess maybe something in the queue is maintaining some state from the previous run.

**Comment by gavin-norman-sociomantic** [comment]
An interesting thing I notice: when the client shuts down, an exception is thrown in the node's connection (as expected):
```
Error [swarm.neo.node.ConnectionHandler] - Exception thrown from request handler: Connection reset by peer (write error) @ ./submodules/swarm/src/swarm/neo/protocol/socket/MessageSender.d:399
```
Then, the `PeriodicRefresh` fiber exits. The other two fibers _do not_ exit, though. This is wrong.

**Comment by gavin-norman-sociomantic** [comment]
There's also a
```
Error [swarm.neo.node.ConnectionHandler] - Exception thrown from request handler: Request unexpectedly running (fiber executing) @ ./submodules/swarm/src/swarm/neo/node/RequestSet.d:255
```
in the mix. This is probably what's messing up the killing of the request fibers.

**Comment by gavin-norman-sociomantic** [comment]
It looks like this is what's happening:
1. A request handler calls some method of the connection which causes the send loop to be resumed.
2. An exception is thrown in the send loop (due to the client shutting down the connection).
3. The connection `shutdownImpl` method is called. This notifies all requests which are registered for sending/receiving of the error.
4. The _node_ connection extends the behaviour of `shutdownImpl` to also kill all active requests by resuming the fibers with an exception.
5. Thus we get to `Request.resumeFiber_` which, in the node's implementation, [asserts that the request fiber being killed is not active](https://github.com/sociomantic-tsunami/swarm/blob/v4.x.x/src/swarm/neo/node/RequestSet.d#L255).
6. BUT this is the fiber which called the send loop in step 1. Thus it _is_ active.

I think this bug is due to a misunderstanding of the fiber states. When fiber A calls fiber B, fiber A remains in state `EXEC`.

**Comment by gavin-norman-sociomantic** [comment]
Fixed by https://github.com/sociomantic-tsunami/swarm/pull/115.

**Comment by gavin-norman-sociomantic** [comment]
Fixed by submodule update in #40.
